tests/frontend/onnx/test_build_ir.py

```diff
@@ -59,3 +62,135 @@ def test_visit_value_info():
+    expected_input2 = """'input2': TestSSAValue(type=TensorType(parameters=(ArrayAttr(data=(IntAttr(data=64),)), Float32Type(parameters=()), NoneAttr(parameters=())), shape=ArrayAttr(data=(IntAttr(data=64),)), element_type=Float32Type(parameters=()), encoding=NoneAttr(parameters=())), _name=None)"""
+    expected_output_pre = "{" + expected_input1 + ", " + expected_input2 + "}"
+
+    assert str(ctx.value_by_name) == expected_output_pre
+
+    # visit node
+    op = visit_node(add_node, ctx)
+
+    assert op.name == "onnx.Add"
+
+    # expected output after visinting the op node
+    expected_out = """'output': <OpResult[tensor<64xf32>] index: 0, operation: onnx.Add, uses: 0>"""
+    expected_output_post = (
+        "{" + expected_input1 + ", " + expected_input2 + ", " + expected_out + "}"
+    )
+
+    assert str(ctx.value_by_name) == expected_output_post
+
+
+def test_visit_node_sub():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create graph composed only of one Sub operation
+    _, sub_node = _create_graph_binary_op("Sub", "sub_graph")
+
+    lhs = TestSSAValue(TensorType(f32, [64]))
+    rhs = TestSSAValue(TensorType(f32, [64]))
+    ctx.value_by_name["input1"] = lhs
+    ctx.value_by_name["input2"] = rhs
+
+    lhs_type = TensorType(f32, [64])
+    rhs_type = TensorType(f32, [64])
+    out_type = TensorType(f32, [64])
+    ctx.type_by_name["input1"] = lhs_type
+    ctx.type_by_name["input2"] = rhs_type
+    ctx.type_by_name["output"] = out_type
+
+    # expected output before visinting the op node
+    expected_input1 = """'input1': TestSSAValue(type=TensorType(parameters=(ArrayAttr(data=(IntAttr(data=64),)), Float32Type(parameters=()), NoneAttr(parameters=())), shape=ArrayAttr(data=(IntAttr(data=64),)), element_type=Float32Type(parameters=()), encoding=NoneAttr(parameters=())), _name=None)"""
+    expected_input2 = """'input2': TestSSAValue(type=TensorType(parameters=(ArrayAttr(data=(IntAttr(data=64),)), Float32Type(parameters=()), NoneAttr(parameters=())), shape=ArrayAttr(data=(IntAttr(data=64),)), element_type=Float32Type(parameters=()), encoding=NoneAttr(parameters=())), _name=None)"""
+    expected_output_pre = "{" + expected_input1 + ", " + expected_input2 + "}"
+
+    assert str(ctx.value_by_name) == expected_output_pre
+
+    # visit node
+    op = visit_node(sub_node, ctx)
+
+    assert op.name == "onnx.Sub"
+
+    # expected output after visinting the op node
+
+    expected_out = """'output': <OpResult[tensor<64xf32>] index: 0, operation: onnx.Sub, uses: 0>"""
+    expected_output_post = (
+        "{" + expected_input1 + ", " + expected_input2 + ", " + expected_out + "}"
+    )
+
+    assert str(ctx.value_by_name) == expected_output_post
```

Comment:
```suggestion
    op = visit_node(sub_node, ctx)

    assert op.is_structurally_equivalent(onnx.Sub(lhs, rhs, out_type))
```
I don't think we need to test the ctx, this test for whether all the operands, results, and attributes are the same should cover everything that we want to be stable, independent of the textual representation of the elements